core/src/main/java/org/lflang/target/TargetConfig.java

```diff
@@ -148,6 +173,71 @@ public TargetConfig(Resource resource, GeneratorArguments args, MessageReporter
     validate(reporter);
   }
 
+  /**
+   * If the federate that target configuration applies to is imported, merge target properties
+   * declared in the file that it was imported from.
+   *
+   * @param importedResource The resource in which the target configuration is to be loaded from.
+   * @param mainResource The resource in which the main reactor is specified.
+   * @param loadOrNot Predicate to determine for each target property whether it should be loaded.
+   * @param messageReporter An error reporter to use when problems are encountered.
+   */
+  public void mergeImportedConfig(
+      Resource importedResource,
+      Resource mainResource,
+      Predicate<TargetProperty> loadOrNot,
+      MessageReporter messageReporter) {
+    // If the federate is imported, then update the configuration based on target properties
+    // in the imported file.
+    if (!importedResource.equals(mainResource)) {
+      var importedTargetDecl = GeneratorUtils.findTargetDecl(importedResource);
+      var targetProperties = importedTargetDecl.getConfig();
+      if (targetProperties != null) {
+        // Merge properties
+        update(
+            this,
+            convertToEmptyListIfNull(targetProperties.getPairs()),
+            FileUtil.getRelativePath(mainResource, importedResource),
+            loadOrNot,
+            messageReporter);
+      }
+    }
+  }
+
+  /**
+   * Update the given configuration using the given target properties.
+   *
+   * @param config The configuration object to update.
+   * @param pairs AST node that holds all the target properties.
+   * @param relativePath The path from the main resource to the resource from which the new
+   *     properties originate.
+   */
+  public void update(
+      TargetConfig config,
+      List<KeyValuePair> pairs,
+      Path relativePath,
+      Predicate<TargetProperty> loadOrNot,
+      MessageReporter err) {
+    pairs.forEach(
+        pair -> {
+          var p = config.forName(pair.getName());
+          if (p.isPresent()) {
+            var value = pair.getValue();
+            var property = p.get();
+            if (property instanceof FileListProperty fileListProperty) {
```

Comment:
Why do we need to handle FileListProperty specially? Is this to update paths that are relative to an imported resource?

Reply:
Exactly.